src/coreclr/vm/riscv64/stubs.cpp

```diff
@@ -1307,151 +1361,370 @@ void StubLinkerCPU::EmitMovReg(IntReg Xd, IntReg Xm)
+        bool isFloating = IsRegisterFloating(lowered0[i].dstofs);
+        TransferredField field = GetTransferredField(lowered0[i].dstofs);
+        if (isFloating)
+        {
+            if (fpRegisters.count > 0 || delowered[0].srcofs != ShuffleEntry::SENTINEL)
+                field.reg = stashFloatReg(field.reg); // the floating field destination is occupied, so stash it
         }
         else
         {
-            // source & dest must be on the stack
-            _ASSERTE(!(pEntry->srcofs & ShuffleEntry::REGMASK));
-            _ASSERTE(!(pEntry->dstofs & ShuffleEntry::REGMASK));
+            _ASSERTE(field.reg == lastIntArgReg); // a7 is vacated so the integer field of lowered0 can go straight home
+        }
+        EmitAnyLoad(isFloating, field.sizeShift, field.reg, RegSp, 0 + field.offset);
+    }
 
-            IntReg t4 = 29;
-            EmitLoad(t4, RegSp, pEntry->srcofs * sizeof(void*));
-            EmitStore(t4, RegSp, pEntry->dstofs * sizeof(void*));
+    // Now that lowered0 is out of the way, we can shuffle subsequent stackSlots0
+    _ASSERTE(stackSlots0.firstSrc + stackSlots0.offset == 0);
+    shuffleStackSlotLeftChain(stackSlots0);
+
+    if (stackSlots1.offset > 0)
+    {
+        // The delowered argument is bigger than lowered0, which displaces subsequent stackSlots1 to the right.
+        // Do the second lowering first to unblock shuffling of stack slots after the delowered argument
+        _ASSERTE(lowered1.srcofs != ShuffleEntry::SENTINEL);
+        TransferredField f = GetTransferredField(lowered1.dstofs);
+        int slotOffset = GetStackSlot(lowered1.srcofs) * sizeof(void*);
+        f.reg = stashFloatReg(f.reg); // destination is occupied, we haven't delowered yet
+        EmitAnyLoad(true, f.sizeShift, f.reg, RegSp, slotOffset + f.offset);
+
+        for (int src = stackSlots1.firstSrc + stackSlots1.count - 1, dst = src + stackSlots1.offset;
+            src >= stackSlots1.firstSrc; --src, --dst)
+        {
+            EmitLoad (intTempReg, RegSp, src * sizeof(void*));
+            EmitStore(intTempReg, RegSp, dst * sizeof(void*));
         }
     }
-    // Tailcall to target
-    // jalr x0, 0(t6)
-    EmitJumpRegister(t6);
+
+    // Now that there's enough room on the stack, delower the argument
+    for (int i = 0; i < ARRAY_SIZE(delowered); ++i)
+    {
+        if (delowered[i].srcofs == ShuffleEntry::SENTINEL)
+            break;
+        _ASSERTE(IsRegisterFloating(delowered[i].srcofs));
+        TransferredField f = GetTransferredField(delowered[i].srcofs);
+        int slotOffset = GetStackSlot(delowered[i].dstofs) * sizeof(void*);
+        EmitAnyStore(true, f.sizeShift, f.reg, RegSp, slotOffset + f.offset);
+    }
+
+    // Last FP regs occupied by the delowered argument have been freed, we can shuffle FP registers to the right
+    for (FloatReg src = fpRegisters.firstSrc + fpRegisters.count - 1;  src >= fpRegisters.firstSrc;  src.reg--)
+    {
+        EmitMovReg(src + 1, src);
```

Comment:
Not covered by EmptyStructs.

Reply:
Right, they used to be covered by the stack growing cases until I switched to IL stubs. Added to ShufflingThunk_PackedEmptyUintEmptyFloat_PackedEmptyDouble.